### PR TITLE
Formatted the code with black

### PR DESCRIPTION
**Pull request by azmeuk**
I used [black](https://github.com/python/black) to format the code.

### CODE REVIEW

File: src/zc/FileStorage/__init__.py [hunk 1]
```diff
@@ -86,46 +87,50 @@ def __init__(self, storage, stop,
...
-                                      ).load()
-                os.remove(self._name+'.packerror')
+            if os.path.exists(self._name + ".packerror"):
+                v = pickle.Unpickler(open(self._name + ".packerror", "rb")).load()
+                os.remove(self._name + ".packerror")
                 raise v
-            raise RuntimeError('The Pack subprocess failed\n'
-                               +'-'*60+out+'-'*60+'\n')
+            raise RuntimeError(
+                "The Pack subprocess failed\n" + "-" * 60 + out + "-" * 60 + "\n"
```
Comment (mgedmin) [inline src/zc/FileStorage/__init__.py:124]: This is one of the rare examples where Black's "no exceptions, just formatting rules" policy makes the code a little bit worse.

Splitting multiline string literals across several lines makes them more readable.

File: src/zc/FileStorage/snapshotintime.py [hunk 2]
```diff
@@ -47,37 +46,36 @@ def main(args=None):
...
         sys.exit(1)
 
     try:
-        date, time = (stop.split('T')+[''])[:2]
-        year, month, day = map(int, date.split('-'))
+        date, time = (stop.split("T") + [""])[:2]
+        year, month, day = map(int, date.split("-"))
         if time:
-            hour, minute, second = (map(int, time.split(':'))+[0,0])[:3]
+            hour, minute, second = (map(int, time.split(":")) + [0, 0])[:3]
```
Comment (mgedmin) [inline src/zc/FileStorage/snapshotintime.py:70]: Unrelated to this PR, but this is a Python 3 compatibility bug -- map() returns an iterator that does not support `__add__`.

This hints to me that this package is maybe unmaintained?

File: src/zc/FileStorage/tests.py [hunk 3]
```diff
@@ -524,21 +527,29 @@ def snapshot_in_time():
 
     """
 
+
 def hexer(data):
-    return (data[:2] == '.h') and data or ('.h'+data.encode('hex'))
+    return (data[:2] == ".h") and data or (".h" + data.encode("hex"))
```
Comment (mgedmin) [inline src/zc/FileStorage/tests.py:532]: (This ought to be rewritten to use the `if .. else` expression.)

File: src/zc/FileStorage/tests.py [hunk 4]
```diff
@@ -524,21 +527,29 @@ def snapshot_in_time():
...
+
+
 def unhexer(data):
-    return data and (data[:2] == '.h' and data[2:].decode('hex') or data)
+    return data and (data[:2] == ".h" and data[2:].decode("hex") or data)
```
Comment (mgedmin) [inline src/zc/FileStorage/tests.py:536]: (Part of this ought to be rewritten to use the `if ... else` expression.)

File: src/zc/FileStorage/__init__.py [hunk 5]
```diff
@@ -86,46 +87,50 @@ def __init__(self, storage, stop,
...
-            syspath = sys.path,
-            blob_dir = self.storage.blob_dir,
-            sleep = self.sleep,
-            transform = self.transform_option,
-            untransform = self.untransform_option,
-            ))
-        for name in 'error', 'log':
-            name = self._name+'.pack'+name
+        script = self._name + ".packscript"
+        open(script, "w").write(
```
Comment (mgedmin) [inline src/zc/FileStorage/__init__.py:91]: (Out of scope for this PR, but all these open()s ought to be embedded in `with` statements.)

**Review by mgedmin — APPROVED** [review]
LGTM!

**Comment by azmeuk** [comment]
Thank you for your reviewing @mgedmin.
I will tackle the points you commented in a new PR if it is ok.
A question on the workflow: how many code review do I need before it is ok to merge the pull request?

**Comment by mgedmin** [comment]
> I will tackle the points you commented in a new PR if it is ok.

Thank you, that would be wonderful!

> A question on the workflow: how many code review do I need before it is ok to merge the pull request?

*vigorous handwave*  ... one?

Given the general state of zopefoundation packages (more on life support rather than being actively maintained) I'd say it's fine to merge without _any_ reviews, if no reviewers show up for a week or so.

Sometimes, when a PR is particularly tricky (e.g. touches core code in zope.interface/zope.component/ZODB), I wouldn't feel comfortable merging my PRs without a review from someone I consider an expert in that package, even if I received reviews from somebody else.

(E.g. I like to do code reviews, but I don't consider myself an expert for core packages.)

**Pull request merged**